03week/ticTacToe.js

```diff
@@ -1,93 +1,248 @@
+  console.log("  ---------");
+  console.log("2 " + board[2].join(" | "));
+};
+
+const isLegalMove = (row, column) => {
+  if (board[row][column] === " ") {
+    return true;
+  } else {
+    console.log("Please pick empty space");
+  }
+};
+
+const playerMove = (row, column) => {
+  if (playerTurn === "X") {
+    board[row][column] = "X";
+  } else {
+    board[row][column] = "O";
+  }
+};
+
+const switchPlayer = () => {
+  if (playerTurn == "X") {
+    playerTurn = "O";
+  } else {
+    playerTurn = "X";
+  }
+};
+
+const horizontalWinX = () => {
+  if (board[0][0] == "X" && board[0][1] == "X" && board[0][2] == "X") {
+    return true;
+  } else if (board[1][0] == "X" && board[1][1] == "X" && board[1][2] == "X") {
+    return true;
+  } else if (board[2][0] == "X" && board[2][1] == "X" && board[2][2] == "X") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const horizontalWinO = () => {
+  if (board[0][0] == "O" && board[0][1] == "O" && board[0][2] == "O") {
+    return true;
+  } else if (board[1][0] == "O" && board[1][1] == "O" && board[1][2] == "O") {
+    return true;
+  } else if (board[2][0] == "O" && board[2][1] == "O" && board[2][2] == "O") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const verticalWinX = () => {
+  if (board[0][0] == "X" && board[1][0] == "X" && board[2][0] == "X") {
+    return true;
+  } else if (board[0][1] == "X" && board[1][1] == "X" && board[2][1] == "X") {
+    return true;
+  } else if (board[0][2] == "X" && board[1][2] == "X" && board[2][2] == "X") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const verticalWinO = () => {
+  if (board[0][0] == "O" && board[1][0] == "O" && board[2][0] == "O") {
+    return true;
+  } else if (board[0][1] == "O" && board[1][1] == "O" && board[2][1] == "O") {
+    return true;
+  } else if (board[0][2] == "O" && board[1][2] == "O" && board[2][2] == "O") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const diagonalWinX = () => {
+  if (board[0][0] == "X" && board[1][1] == "X" && board[2][2] == "X") {
+    return true;
+  } else if (board[0][2] == "X" && board[1][1] == "X" && board[2][0] == "X") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const diagonalWinO = () => {
+  if (board[0][0] == "O" && board[1][1] == "O" && board[2][2] == "O") {
+    return true;
+  } else if (board[0][2] == "O" && board[1][1] == "O" && board[2][0] == "O") {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const checkForWin = () => {
+  if (horizontalWinX() || verticalWinX() || diagonalWinX()) {
+    console.log("Player X Wins!");
+    return true;
+  } else if (horizontalWinO() || verticalWinO() || diagonalWinO()) {
+    console.log("Player O Wins!");
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const resetGame = () => {
+  board = [[" ", " ", " "], [" ", " ", " "], [" ", " ", " "]];
+  playerTurn = "X";
+};
+
+const ticTacToe = (row, column) => {
+  if (isLegalMove) {
```

Comment:
Your isLegalMove is always returning true even if the spot is already taken. This has to do with calling the function && passing the arguments and referencing a function."referencing meaning that does it exists."